src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php

```diff
@@ -242,9 +246,12 @@ public function endTest($test, $time)
         $className = \get_class($test);
         $groups = Test::getGroups($className, $test->getName(false));
 
-        if ($errored = null !== $this->error) {
-            $test->getTestResultObject()->addError($test, $this->error, 0);
-            $this->error = null;
+        if ($this->checkNumAssertions) {
+            if (!self::$expectedDeprecations && !$test->getNumAssertions()) {
```

Comment:
I slightly changed the code here to always reset `checkNumAssertions` to false when a test set it to `true`.